### ISSUE TITLE
shiftwidth appears to be changed by loading a workspace

### ISSUE THREAD

**Issue by mellery451** [vim-ctrlspace/vim-ctrlspace#203]
I usually work with a shiftwidth of 4 (set in my rc file).  Here's what I've noticed:

if I open `vim somefile`, shiftwidth is 4.
if I launch vim and then open `somefile`, shiftwidth is 4
if I launch vim, invoke ctrl-space-w and load my main workspace and then select `somefile` from the buffers list, shiftwidth is 2. 

I've tried this several times to convince myself that the loading of a workspace seems to be mucking with shiftwidth. I have tried with latest versions of vim and neovim -- both repro the same. 

Any thoughts about how this could happen? Does workspace loading ever touch shiftwidth (I could not see anything obvious in the code)? Maybe something to do with the buffer window that opens when ctrl-space is active?

**Comment by cometsong** [comment]
As I have my shiftwidth and use of tabs/spaces differing between filetypes, e.g. Makefiles, python, bash, etc. then I propose a testing scenario below.

How it's pre-defined in your vimrc and if the syntax or ftplugin may reset it after loading (though you've already tested that it's only* after* loading the workspace, *that* action itself may load things in diffrerent order than the usual vim editing new files.)  

For reference, the ctrlspace code loads the workspace using `session` info native to vim; so that depends on which [n]vim versions you're using, as we've discovered recently there have been some changes in the base vim behaviours (re: #202).

To test: try opening the buffers as you mentioned above and use `:mksession load_test.vim` and then quit and reopen vim using command `vim -S load_test.vim`.  This at least will check if it's the session loading (as I said, version-specific) that does that.

**Comment by mellery451** [comment]
good suggestion. I tried creating/loading the session with `:mksession` and `shiftwidth` is correctly still 4 after loading. Just to sanity check, without exiting I then did ctrlspace-w to load my workspace and `shiftwidth` is now 2 for the same file. 

This test was done with nvim 0.3.1. I tried the same with vim 8.1 and although the `shiftwidth` seemed to be correct before/after loading, but I also get lots of errors from various plugins (like ale and fugitive) after reload - I'm not sure if this is something wonky with my mappings or something that's just off with vim 8 + sessions. In any event, nvim is perfectly happy reloading a session and that's what I'm primarily using anyhow.

I feel like this `shiftwidth` thing is somehow related to the window that ctrlspace creates for navigating buffers,sessions,tabs,etc, but I'll admit I can't actually spot anywhere in the plugin code that the value is actually being messed with.
